app/api/crud.py

```diff
@@ -87,7 +87,9 @@ async def get(
                 AggDatasetResponse(
                     dataset=dataset,
                     dataset_name=dataset_name,
-                    num_matching_subjects=group.shape[0],
+                    num_matching_subjects=group["sub_id"].unique().shape[0],
+                    subject_file_paths=list(group["file_path"]),
+                    image_modals=list(group["image_modal"].unique()),
```

Comment:
I like having the info about the represented image modalities per dataset, but I think it could be a bit clearer either on the API side or whenever you implement it on the frontend side. For instance, right now a value of
```
"image_modals": ["http://purl.org/nidash/nidm#T1Weighted", "http://purl.org/nidash/nidm#T2Weighted"]
```
for a no-filters query means that at least one subject in each returned dataset has images of each listed modality. I think this output becomes unintuitive to interpret for datasets where not everyone has the same modalities available (e.g., if 100/100 subjects have T1 images, but only 50/100 have T2 images), or for queries where you're filtering by a certain modality. e.g., If a user is just looking for subjects with T1 data, the resulting `image_modals` could still be the same as the above which means that all subjects returned have T1 data, and _at least one_ of those subjects also has T2 data. But I'm not sure this would be immediately clear from the `image_modals` list itself.

I realize that currently the test data provides only datasets where everyone has the same modalities acquired, so this could definitely be a discussion point to revisit.

Some ideas to make this output more intuitive:
- Could we combine the aggregate image modality info with the number of matching subjects? To show the _# subjects per modality_ for a given query.
- We could just document clearly how to interpret the `image_modals` list (i.e., what multiple modalities means when you queried for only one, or the fact that it doesn't necessarily apply to all matching subjects), either in the frontend or in user-facing documentation.

What do you think?

Reply:
Yea I agree with you, I think we can document it for the first release and revisit it after.

Reply:
I agree with both of you :) ! @alyssadai do you mind spinning your comment off into its own issue [like so](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue#creating-an-issue-from-a-comment) and then we can continue discuss things there?